crates/loam-cli/examples/soroban/nft/src/test.rs

```diff
@@ -0,0 +1,263 @@
+#![cfg(test)]
+
+use super::{
+    SorobanContract__ as SorobanContract, SorobanContract__Client as SorobanContractClient,
+};
+use loam_sdk::soroban_sdk::{
+    testutils::{Address as _, MockAuth, MockAuthInvoke},
+    Address, Bytes, Env, IntoVal,
+};
+
+extern crate std;
+
+mod contract {
+    use loam_sdk::soroban_sdk;
+
+    soroban_sdk::contractimport!(file = "../../../../../target/loam/example_nft.wasm");
+}
+
+fn init() -> (Env, SorobanContractClient<'static>, Address) {
+    let env = Env::default();
+    let contract_id = env.register_contract(None, SorobanContract);
+    let client = SorobanContractClient::new(&env, &contract_id);
+    (env, client, contract_id)
+}
+
+fn set_admin(env: &Env, client: &SorobanContractClient, contract_id: &Address, admin: &Address) {
+    client
+        .mock_auths(&[MockAuth {
+            address: admin,
+            invoke: &MockAuthInvoke {
+                contract: contract_id,
+                fn_name: "admin_set",
+                args: (admin,).into_val(env),
+                sub_invokes: &[],
+            },
+        }])
+        .admin_set(admin);
+}
+
+fn init_nft_contract(
+    env: &Env,
+    client: &SorobanContractClient,
+    contract_id: &Address,
+    admin: &Address,
+    name: &str,
+) {
+    let name = Bytes::from_slice(env, name.as_bytes());
+    client
+        .mock_auths(&[MockAuth {
+            address: &admin,
+            invoke: &MockAuthInvoke {
+                contract: &contract_id,
+                fn_name: "nft_init",
+                args: (name.clone(),).into_val(env),
+                sub_invokes: &[],
+            },
+        }])
+        .nft_init(&name);
+}
+
+fn mint_nft(
+    env: &Env,
+    client: &SorobanContractClient,
+    contract_id: &Address,
+    admin: &Address,
+    owner: &Address,
+    metadata: &Bytes,
+) -> u128 {
+    client
+        .mock_auths(&[MockAuth {
+            address: &admin,
+            invoke: &MockAuthInvoke {
+                contract: &contract_id,
+                fn_name: "mint",
+                args: (owner.clone(), metadata.clone()).into_val(env),
+                sub_invokes: &[],
+            },
+        }])
+        .mint(&owner, &metadata)
+}
+
+fn transfer_nft(
+    env: &Env,
+    client: &SorobanContractClient,
+    contract_id: &Address,
+    nft_id: u128,
+    owner: &Address,
+    new_owner: &Address,
+) {
+    client
+        .mock_auths(&[MockAuth {
+            address: &owner,
+            invoke: &MockAuthInvoke {
+                contract: &contract_id,
+                fn_name: "transfer",
+                args: (nft_id.clone(), owner.clone(), new_owner.clone()).into_val(env),
+                sub_invokes: &[],
+            },
+        }])
+        .transfer(&nft_id, &owner, &new_owner);
+}
+
+#[test]
+fn test_nft() {
```

Comment:
While testing with Env, I had trouble getting the tests to pass while running in parallel. ChatGpt told me that the errors could be caused by a "context already borrowed" issue. It kind of feels like the env isn't being reset in between tests, even though I am presumably creating a new env for each test. 

These tests do pass when I run them with just one thread:
```
cargo test --package example-nft --lib -- test --test-threads=1
```

@willemneal have you experienced anything like this while testing with env?